packages/cluster-ui/src/transactionsPage/transactionsPage.tsx

```diff
@@ -74,11 +69,17 @@ export class TransactionsPage extends React.Component<
       pageSize: this.props.pageSize || 20,
       current: 1,
     },
-    search: this.trxSearchParams("q", ""),
+    search: this.trxSearchParams("q", "").toString(),
```

Comment:
With all the type conversion here, I wonder if we need to change this approach. Since we have abstracted the filters, I wonder if we could have something like a `getFiltersFromQueryString` function instead.

```js
const getFilters = (defaults, queryString) => {
  // create search params from query string
  const searchParams = new URLSearchParams(qs);
  
  // for each default value, look on the query string for an equivelant key
  return Object.keys(defaults).reduce((filters, filter) => {
    // get default value for filter
    const defaultValue = defaults[filter];
    // get filter from the search params, if it exists
    const queryStringFilter = searchParams.get(filter);
    // This part might look a little tricky. First we determine if the queryStringFilter is
    // null (the value returned for an absent filter). If it is, we simply return the default
    // value for the filter. If it isn't we use the constructor of the value of
    // the default filter (that is the function used to create it, e.g. Number, String, Boolean)
    // to coerse the string from the search params into the some type as the default
    const filterValue = (queryStringFilter === null) ? defaultValue : (defaultValue.constructor)(searchParams.get(filter));
    // add the value to the reduced object
    return { [filter]: filterValue, ...filters };
  }, {}) 
}
```

I'll admit, that example got a little out of control. But the tricky part is getting the filter from the query string (which is a string) to be the same type as the value in the defaults. I also didn't try to run this through TypeScript, so let me know if you want to pair on trying to solve this.